collector/nfsd_linux.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+	"github.com/prometheus/procfs/nfs"
+)
+
+// A nfsdCollector is a Collector which gathers metrics from /proc/net/rpc/nfsd.
+// See: https://www.svennd.be/nfsd-stats-explained-procnetrpcnfsd/
+type nfsdCollector struct {
+	fs procfs.FS
+}
+
+func init() {
+	registerCollector("nfsd", defaultEnabled, NewNFSdCollector)
+}
+
+const (
+	nfsdSubsystem = "nfsd"
+)
+
+var (
+	nfsdReplyCacheHitsDesc = prometheus.NewDesc(
```

Comment:
Nit, but I'd make these fields on the `nfsdCollector` type so you don't pollute the package level with variables.

Reply:
Ok, I'll move things around.